api/schema/attributemode.go

```diff
@@ -34,6 +34,17 @@ type AttributeMode struct {
 var (
 	// These variables should be considered constant and must not be modified.
 
+	// CreatableIDAttributeMode is the mode of "id" attributes that are passed in creation.
+	CreatableIDAttributeMode = AttributeMode{
```

Comment:
How about "Key" instead of "Creatable"? That would convey that this is used to identify the resource.
```suggestion
	// KeyAttributeMode is the mode of attributes that are used to identify resource instances.
	KeyAttributeMode = AttributeMode{
```